src/lib/ops_and_meta_predicates.pl

```diff
@@ -5,9 +5,13 @@
 
 :- op(1199, fx, meta_predicate).
 
-/* this is an implementation specific declarative operator used to implement call_with_inference_limit/3
-   and setup_call_cleanup/3. switches to the default trust_me and retry_me_else. Indexing choice
-   instructions are unchanged. */
+% Declarative operator used to implement `call_with_inference_limit/3` and
```

Comment:
May I suggest to please put such drastic documentation changes into their own PR so that they can be discussed separately? For instance, is there any reason to remove "implementation specific" from the existing description? The standard defines *implementation specific* as:

<pre>
<b>3.92 implementation specific</b>: Undefined by this part of
ISO/IEC 13211 but supported by a conforming processor.
</pre>

So, this operator is really implementation specific, which is conforming:

<pre>
<b>5.5.2 Predefined operators</b>

A processor may support one or more additional predefined
operators (table 7) as an implementation specific feature.
</pre>

In this concrete cause, the issue worth discussing is whether `non_counted_backtracking` should even *be* an operator. It seems to be used only in the Scryer codebase itself, and there seems no need to add a dedicated operator just for that purpose, because it must not appear in a **strictly&nbsp;conforming** execution mode (which should be the default mode in Scryer):

<pre>
<b>5.1 Prolog processor</b>

A conforming Prolog processor shall:

...

  e) Offer a <b>strictly conforming mode</b> which shall reject
  the use of an implementation specific feature in Prolog
  text or while executing a goal.
</pre>

Reply:
Thanks, it looks like it is not a "small fix". I'll create a separate PR for this change, I think this operator needs better documentation, because for me it was not obvious at all if I should mark my new predicates inside loader.pl as having "non counted backtracking" or not.

Reply:
New PR created #2534